pkg/apis/networking/v1alpha1/ingress_lifecycle_test.go

```diff
@@ -112,3 +114,34 @@ func TestIngressTypicalFlow(t *testing.T) {
 	r.MarkIngressNotReady("", "")
 	apitest.CheckConditionOngoing(r.duck(), IngressConditionReady, t)
 }
+
+func TestIngressGetCondition(t *testing.T) {
+	ingressStatus := &IngressStatus{}
+	ingressStatus.InitializeConditions()
+	tests := []struct {
+		name     string
+		condType apis.ConditionType
+		expect   *apis.Condition
+	}{{
+		name:     "random condition",
+		condType: apis.ConditionType("random"),
+		expect:   nil,
+	}, {
+		name:     "ready condition",
+		condType: apis.ConditionReady,
+		expect: &apis.Condition{
+			Status: corev1.ConditionUnknown,
+		},
+	}, {
+		name:     "succeeded condition",
+		condType: apis.ConditionSucceeded,
+		expect:   nil,
+	}}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			if got, want := ingressStatus.GetCondition(tc.condType), tc.expect; got != nil && got.Status != want.Status {
```

Comment:
I think you need deep compare statuses?

Reply:
Statuses are of type string so i think deep compare not required here

`Status corev1.ConditionStatus`
```
type ConditionStatus string

// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
// can't decide if a resource is in the condition or not. In the future, we could add other
// intermediate conditions, e.g. ConditionDegraded.
const (
	ConditionTrue    ConditionStatus = "True"
	ConditionFalse   ConditionStatus = "False"
	ConditionUnknown ConditionStatus = "Unknown"
)
```